tm2/pkg/p2p/switch_test.go

```diff
@@ -890,3 +890,36 @@ func TestCalculateBackoff(t *testing.T) {
 		}
 	})
 }
+
+func TestSwitchAcceptLoopTransportClosed(t *testing.T) {
+	t.Parallel()
+
+	var transportClosed bool
+
+	ccAccept := make(chan struct{})
+	mockTransport := &mockTransport{
+		acceptFn: func(context.Context, PeerBehavior) (PeerConn, error) {
+			if !transportClosed {
+				transportClosed = true
+				return nil, errTransportClosed
+			}
+
+			ccAccept <- struct{}{}
+			return nil, errors.New("shouldn't reach here")
+		},
+	}
+
+	sw := NewMultiplexSwitch(mockTransport)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Run the accept loop
+	go sw.runAcceptLoop(ctx)
+
+	select {
+	case <-time.After(time.Millisecond * 500): // wait enough time
```

Comment:
You should have a channel that fires in the mock func for the successful case (transportClosed = true), and wait here. The other case is a time.After that times out. This also means you'd need to check that `transportClosed` is `true`

Reply:
Just a suggestion, I think it's easier to follow along this way

Reply:
I've updated the logic so it does not have to wait for a timeout and correctly asserts if the `transportClose` is true.